### PR TITLE
add a init program

### PR DESCRIPTION
**Pull request by xudifsd**
Some benefit of init program:

* reap zombie processes
* capture signal and print out for debug
* capture exit signal and broadcast to user's process, allow graceful exit, also improve exit speed

compiled init program has 2.2MB, and takes 5s to terminate the pod instead of 30s.

Before this, integration test takes 3m56s:
```
real    3m56.608s
user    0m11.417s
sys     0m2.814s
```

After this, takes 2m35s:
```
real    2m35.153s
user    0m10.407s
sys     0m2.847s
```

### CODE REVIEW

File: src/init-scripts/bootstrap.sh [hunk 1]
```diff
@@ -39,15 +39,9 @@ set +e
...
 if [ "$DLTS_ROLE_NAME" = "worker" ];
 then
-    runuser -l ${DLTS_USER_NAME} -c "sleep infinity"
+    exec /usr/local/bin/init runuser -l ${DLTS_USER_NAME} -c "sleep infinity"
 else
     printenv DLTS_LAUNCH_CMD > /dlts-runtime/status/job_command.sh
     chmod +x /dlts-runtime/status/job_command.sh
-    runuser -s /bin/bash -l ${DLTS_USER_NAME} -c /dlts-runtime/status/job_command.sh
-    # Save exit code
-    EXIT_CODE=$?
```
Comment (Anbang-Hu) [inline src/init-scripts/bootstrap.sh:48]: We don't need `EXIT_CODE` any more?

File: src/docker-images/init-container/runtime/init.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	cmd := exec.Command(command[0], command[1:]...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+
+	go func() {
+		for sig := range sigs {
+			if sig != syscall.SIGCHLD {
+				fmt.Fprintf(os.Stderr, "[init] received signal %d\n", sig)
+				syscall.Kill(-cmd.Process.Pid, sig.(syscall.Signal))
```
Comment (Anbang-Hu) [inline src/docker-images/init-container/runtime/init.go:82]: Not familiar. Just want to confirm `-` before pid is needed?

**Review by Anbang-Hu — APPROVED** [review]

File: src/docker-images/init-container/runtime/init.go (hunk 2 above)
Comment (xudifsd) [reply src/docker-images/init-container/runtime/init.go:82]: Yes, this means send signal to that process group. We let Command use a new process group before run it.

File: src/init-scripts/bootstrap.sh (hunk 1 above)
Comment (xudifsd) [reply src/init-scripts/bootstrap.sh:48]: Yes, because we use `exec`, means no bash code will be executed after it.

Comment (xudifsd) [reply src/init-scripts/bootstrap.sh:48]: The init process will handle this

**Pull request merged**